.devcontainer/Dockerfile

```diff
@@ -0,0 +1,24 @@
+# See here for image contents: https://github.com/microsoft/vscode-dev-containers/tree/v0.209.6/containers/python-3/.devcontainer/base.Dockerfile
+
+# [Choice] Python version (use -bullseye variants on local arm64/Apple Silicon): 3, 3.10, 3.9, 3.8, 3.7, 3.6, 3-bullseye, 3.10-bullseye, 3.9-bullseye, 3.8-bullseye, 3.7-bullseye, 3.6-bullseye, 3-buster, 3.10-buster, 3.9-buster, 3.8-buster, 3.7-buster, 3.6-buster
+ARG VARIANT="3.10-bullseye"
+FROM mcr.microsoft.com/vscode/devcontainers/python:0-${VARIANT}
+
+# [Choice] Node.js version: none, lts/*, 16, 14, 12, 10
+ARG NODE_VERSION="none"
+RUN if [ "${NODE_VERSION}" != "none" ]; then su vscode -c "umask 0002 && . /usr/local/share/nvm/nvm.sh && nvm install ${NODE_VERSION} 2>&1"; fi
+
+# [Optional] If your pip requirements rarely change, uncomment this section to add them to the image.
+# COPY requirements.txt /tmp/pip-tmp/
+# RUN pip3 --disable-pip-version-check --no-cache-dir install -r /tmp/pip-tmp/requirements.txt \
+#    && rm -rf /tmp/pip-tmp
+
+# [Optional] Uncomment this section to install additional OS packages.
+# RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
+#     && apt-get -y install --no-install-recommends <your-package-list-here>
+
+# [Optional] Uncomment this line to install global node packages.
+# RUN su vscode -c "source /usr/local/share/nvm/nvm.sh && npm install -g <your-package-here>" 2>&1
+
+# The following code sets up DDEV's "mkdoc" environment
+RUN pip install mkdocs mkdocs-material mdx_truly_sane_lists mkdocs-git-revision-date-localized-plugin
```

Comment:
I don't understand why we're installing these few things instead of installing everything in requirements.txt?

Also don't understand using pip instead of pip3?

Reply:
I'm still learning docker & how to best intergrate it into my workflow.

Instead of installing the requirements for mkdocs in WSL, I thought I'd try the VSCode remote container to install them.
As I hit the errors listed below, I updated the `RUN pip` line to resolve them:

```
ERROR    -  Config value: 'theme'. Error: Unrecognised theme name: 'material'. The available installed themes are: mkdocs, readthedocs
ERROR    -  Config value: 'markdown_extensions'. Error: Failed loading extension "mdx_truly_sane_lists".
ERROR    -  Config value: 'plugins'. Error: The "git-revision-date-localized" plugin is not installed
```

For my setup, this PR spun up a working mkdocs server. After reading through your comments though, I see my process has several flaws.